.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,12 +1,22 @@
-<!--
-Please follow the naming conventions in the "guidelines for contributing" link above or the pull request may be rejected.
+Fixes #
 
-Also, do consider outlining the solution taken as doing so will likely help in the reviewing process
-(e.g., when the pull request involves non-trivial changes)
--->
+**PR Checklist**
 
-Fixes #
+<!-- Remove this portion after you have made the checks. -->
+
+Have you:
+- [ ] Read and understood [our PR guideline (steps 4-6)](https://github.com/TEAMMATES/teammates/blob/master/docs/process.md#step-4-submit-a-pr)?
+  - [ ] Added the issue number to the "Fixes" keyword above?
+  - [ ] Titled the PR as specified in the abovementioned document?
+- [ ] Made your changes on a branch other than `master`?
+- [ ] Used good commit messages?
```

Comment:
"Good" needs to be specified more concretely. I just realised that we don't have a standard or guidelines for contributor to follow (the reference is hidden in a link--I missed it on the first pass even though I was looking for one). Perhaps we can start moving towards adopting the oss-generic guidelines for [messages](https://github.com/oss-generic/process/blob/master/docs/FormatsAndConventions.md#commit-message) and [organisation](https://github.com/oss-generic/process/blob/master/docs/FormatsAndConventions.md#commit-organization)? It may be a little extra work on the contributor's part but will make the reviewers' lives a lot easier. Especially with the influx of new contributors.